### PR TITLE
Refactor: Unexplode Backup() function, pass BackupRequest as argument

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Similarly to https://github.com/vitessio/vitess/pull/10578 and https://github.com/vitessio/vitess/pull/10671, this PR unexplodes `Backup()` functions to include a `BackupRequest` instead of specific request fields like `concurrency` and `allowPrimary`.

I have used both `vtctldatapb.BackupRequest` as well as `tabletmanagerdata.BackupRequest`. I hope I've used both in their appropriate context - kindly consider this while reviewing.

## Related Issue(s)

#10578 , #10671 

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtctl/grpcvtctldserver/testutil/test_tmclient.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ import (
 	logutilpb "vitess.io/vitess/go/vt/proto/logutil"
 	querypb "vitess.io/vitess/go/vt/proto/query"
 	replicationdatapb "vitess.io/vitess/go/vt/proto/replicationdata"
+	"vitess.io/vitess/go/vt/proto/tabletmanagerdata"
```
Comment (ajm188) [inline go/vt/vtctl/grpcvtctldserver/testutil/test_tmclient.go:43]: this is already imported (directly below) as `tabletmanagerdatapb`

File: go/vt/vtctl/grpcvtctldserver/server.go [hunk 2]
```diff
@@ -386,13 +386,15 @@ func (s *VtctldServer) BackupShard(req *vtctldatapb.BackupShardRequest, stream v
...
+	r := &vtctldatapb.BackupRequest{Concurrency: req.Concurrency, AllowPrimary: req.AllowPrimary}
+	return s.backupTablet(ctx, backupTablet, r, stream)
 }
 
-func (s *VtctldServer) backupTablet(ctx context.Context, tablet *topodatapb.Tablet, concurrency int, allowPrimary bool, stream interface {
+func (s *VtctldServer) backupTablet(ctx context.Context, tablet *topodatapb.Tablet, req *vtctldatapb.BackupRequest, stream interface {
 	Send(resp *vtctldatapb.BackupResponse) error
 }) error {
-	logStream, err := s.tmc.Backup(ctx, tablet, concurrency, allowPrimary)
+	r := &tabletmanagerdatapb.BackupRequest{Concurrency: int64(req.Concurrency), AllowPrimary: req.AllowPrimary}
```
Comment (ajm188) [inline go/vt/vtctl/grpcvtctldserver/server.go:396]: 👍 per "i hope i've used each in their appropriate context", i agree with this delineation

File: go/vt/vttablet/faketmclient/fake_client.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ import (
 	logutilpb "vitess.io/vitess/go/vt/proto/logutil"
 	querypb "vitess.io/vitess/go/vt/proto/query"
 	replicationdatapb "vitess.io/vitess/go/vt/proto/replicationdata"
+	"vitess.io/vitess/go/vt/proto/tabletmanagerdata"
```
Comment (ajm188) [inline go/vt/vttablet/faketmclient/fake_client.go:39]: already imported as `tabletmanagerdatapb`

File: go/vt/vttablet/grpctmclient/client.go [hunk 4]
```diff
@@ -41,6 +41,7 @@ import (
 	logutilpb "vitess.io/vitess/go/vt/proto/logutil"
 	querypb "vitess.io/vitess/go/vt/proto/query"
 	replicationdatapb "vitess.io/vitess/go/vt/proto/replicationdata"
+	"vitess.io/vitess/go/vt/proto/tabletmanagerdata"
```
Comment (ajm188) [inline go/vt/vttablet/grpctmclient/client.go:44]: imports

File: go/vt/vttablet/tabletmanager/rpc_backup.go [hunk 5]
```diff
@@ -27,6 +27,7 @@ import (
 	"vitess.io/vitess/go/vt/topo/topoproto"
 	"vitess.io/vitess/go/vt/vterrors"
 
+	"vitess.io/vitess/go/vt/proto/tabletmanagerdata"
```
Comment (ajm188) [inline go/vt/vttablet/tabletmanager/rpc_backup.go:30]: can you change this to match the other aliased proto imports? (should just append `pb`)

File: go/vt/vttablet/tmclient/rpc_client_api.go [hunk 6]
```diff
@@ -29,6 +29,7 @@ import (
 
 	querypb "vitess.io/vitess/go/vt/proto/query"
 	replicationdatapb "vitess.io/vitess/go/vt/proto/replicationdata"
+	"vitess.io/vitess/go/vt/proto/tabletmanagerdata"
```
Comment (ajm188) [inline go/vt/vttablet/tmclient/rpc_client_api.go:32]: imports

File: go/vt/vttablet/tmrpctest/test_tm_rpc.go [hunk 7]
```diff
@@ -40,6 +40,7 @@ import (
 
 	querypb "vitess.io/vitess/go/vt/proto/query"
 	replicationdatapb "vitess.io/vitess/go/vt/proto/replicationdata"
+	"vitess.io/vitess/go/vt/proto/tabletmanagerdata"
```
Comment (ajm188) [inline go/vt/vttablet/tmrpctest/test_tm_rpc.go:43]: imports

**Review by ajm188 — COMMENTED** [review]
lgtm, pending the import duplication/naming

**Comment by shlomi-noach** [comment]
@ajm188 fixed all duplicate imports. Nice catch!

File: go/vt/vtcombo/tablet_map.go [hunk 8]
```diff
@@ -33,6 +31,7 @@ import (
 	"vitess.io/vitess/go/vt/logutil"
 	"vitess.io/vitess/go/vt/mysqlctl"
 	"vitess.io/vitess/go/vt/mysqlctl/tmutils"
+	"vitess.io/vitess/go/vt/proto/vschema"
```
Comment (ajm188) [inline go/vt/vtcombo/tablet_map.go:34]: for consistency, this should probably be `vschemapb` (and with the other `/proto/` imports below) (sorry i missed this one last time!)

**Review by ajm188 — APPROVED** [review]
lgtm, with the other comment i left (feel free to not block on it if you don't want to!)

File: go/vt/vtcombo/tablet_map.go (hunk 8 above)
Comment (shlomi-noach) [reply go/vt/vtcombo/tablet_map.go]: `vschemapb` it is

**Comment by shlomi-noach** [comment]
`Cluster (topo_connection_cache)` consistently [fails](https://github.com/vitessio/vitess/runs/7624410926?check_suite_focus=true) on what seems to be related to backups. I'm unsure yet how this PR affects the test, but apparently it does.

**Comment by shlomi-noach** [comment]
Still clueless on the failing test.

**Comment by shlomi-noach** [comment]
huh. After merging `main`, `topo_connection_cache` now passed

**Pull request merged**